File: HomeEnergyUsageTracker/src/HomeEnergyUsageTracker.Core/Model/SavingsTipAggregate/SavingsTip.cs
// Licensed to the .NET Foundation under one or more agreements.
// The .NET Foundation licenses this file to you under the MIT license.

namespace HomeEnergyUsageTracker.Core;

public class SavingsTip
{
    public Guid SavingsTipId { get; set; }

    /// <summary>
    /// Gets or sets the tenant ID for multi-tenant isolation.
    /// </summary>
    public Guid TenantId { get; set; }
    public string Title { get; set; } = string.Empty;
    public string? Description { get; set; }
    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
}

File: HomeEnergyUsageTracker/src/HomeEnergyUsageTracker.WebApp/projects/home-energy-usage-tracker/src/app/models/savings-tip.ts
export interface SavingsTip {
  savingsTipId: string;
  title: string;
  description?: string;
  createdAt: Date;
}

export interface CreateSavingsTipRequest {
  title: string;
  description?: string;
}

export interface UpdateSavingsTipRequest {
  savingsTipId: string;
  title: string;
  description?: string;
}
